src/server/src/main/java/io/cassandrareaper/management/http/HttpMetricsProxy.java

```diff
@@ -18,48 +18,187 @@
+                           Supplier<OkHttpClient> httpClientSupplier) {
     this.proxy = proxy;
+    this.node = node;
+    this.httpClient = httpClientSupplier.get();
+  }
+
+  public static HttpMetricsProxy create(HttpCassandraManagementProxy proxy, Node node) {
+    return new HttpMetricsProxy(proxy, node, () -> new OkHttpClient.Builder().build());
   }
 
-  public static HttpMetricsProxy create(HttpCassandraManagementProxy proxy) {
-    return new HttpMetricsProxy(proxy);
+  @VisibleForTesting
+  static HttpMetricsProxy create(
+      HttpCassandraManagementProxy proxy, Node node, Supplier<OkHttpClient> httpClientSupplier) {
+    return new HttpMetricsProxy(proxy, node, httpClientSupplier);
   }
 
   @Override
-  public List<GenericMetric> collectTpStats(Node node) throws JMException, IOException {
-    throw new UnsupportedOperationException("Not implemented");
+  public List<GenericMetric> collectTpStats() throws JMException, IOException {
+    // Collect all metrics and filter out the ones that are not thread pool metrics
+    List<GenericMetric> genericMetrics = MetricsProxy.convertToGenericMetrics(collectMetrics(), node);
+    return genericMetrics.stream()
+        .filter(metric -> metric.getMetricType().equals("ThreadPools"))
+        .collect(Collectors.toList());
   }
 
   @Override
-  public List<GenericMetric> collectDroppedMessages(Node node) throws JMException, IOException {
-    throw new UnsupportedOperationException("Not implemented");
+  public List<GenericMetric> collectDroppedMessages() throws JMException, IOException {
+    // Not mandatory for our first iteration.
+    return Collections.emptyList();
   }
 
   @Override
-  public List<GenericMetric> collectLatencyMetrics(Node node) throws JMException, IOException {
-    throw new UnsupportedOperationException("Not implemented");
+  public List<GenericMetric> collectLatencyMetrics() throws JMException, IOException {
+    // Not mandatory for our first iteration.
+    return Collections.emptyList();
   }
 
   @Override
-  public List<GenericMetric> collectGenericMetrics(Node node) throws JMException, IOException {
-    throw new UnsupportedOperationException("Not implemented");
+  public List<GenericMetric> collectGenericMetrics() throws JMException, IOException {
+    // Not mandatory for our first iteration.
+    return Collections.emptyList();
   }
 
   @Override
-  public List<GenericMetric> collectPercentRepairedMetrics(Node node, String keyspaceName)
+  public List<GenericMetric> collectPercentRepairedMetrics(String keyspaceName)
       throws JMException, IOException {
     throw new UnsupportedOperationException("Not implemented");
   }
+
+  private Map<String, List<JmxStat>> collectMetrics() throws IOException {
+    // TODO: add a short lived cache to avoid querying the metrics endpoint too often
+    Request request = new Request.Builder()
+        .url("http://" + proxy.getHost() + ":" + proxy.getMetricsPort() + "/metrics")
```

Comment:
issue: This will cause serious load on Reaper node (and the Cassandra node) as it will transfer all the metrics through network. This is the reason the filtering method was created, where you can append the name filtering to get the exact metric (and nothing else).

So the syntax is: ``http://localhost:9000/metrics?name=org_apache_metrics_threadetc`` and that would return only the org_apache_metrics_threadetc metric (exact name is required).

Reply:
I think this has been resolved by allowing the method to take an optional of the metric name and filter on that.